connectivity/tests/errors.go

```diff
@@ -20,29 +21,47 @@ import (
 	"github.com/cilium/cilium-cli/utils/features"
 )
 
+type logMatcher interface {
+	IsMatch(log string) bool
+}
+
+type stringMatcher string
+
+func (s stringMatcher) IsMatch(log string) bool {
+	return strings.Contains(log, string(s))
+}
+
+type regexMatcher struct {
+	*regexp.Regexp
+}
+
+func (r regexMatcher) IsMatch(log string) bool {
+	return r.Regexp.Match([]byte(log))
```

Comment:
nit: could use `regexp.(*Regexp).MatchString` here:

```suggestion
	return r.Regexp.MatchString(log)
```